### PR TITLE
add big zone retirement

### PR DESCRIPTION
**Pull request by struct**
Add support for big zone retirement with compile time define configuration

### CODE REVIEW

**Comment by jvoisin** [comment]
Doesn't this provide a straightforward DoS vector?

File: include/iso_alloc_ds.h [hunk 1]
```diff
@@ -68,6 +68,7 @@ typedef struct iso_alloc_big_zone_t {
     uint64_t canary_a;
     bool free;
     uint64_t size;
+    uint32_t ttl;
```
Comment (jvoisin) [inline include/iso_alloc_ds.h:71]: An `unsigned char` should be enough I think.

**Comment by struct** [comment]
How so?

**Comment by jvoisin** [comment]
Oh, I misunderstood: they're retired as in "destroyed and given back to the OS", not as in "permanently put into quarantine", my bad.

What's the rationale for re-using them in the first place by the way?

**Comment by struct** [comment]
> What's the rationale for re-using them in the first place by the way?

Performance mainly, otherwise any big zone allocation requires multiple syscalls. The current state is definitely not final. Im not very happy with the overall memory usage and perf numbers but we can't check all boxes for all workloads.

**Pull request merged**